docs/Reference/Hooks.md

```diff
@@ -267,6 +268,26 @@ service (if the `connectionTimeout` property is set on the Fastify instance).
 The `onTimeout` hook is executed when a request is timed out and the HTTP socket
 has been hanged up. Therefore, you will not be able to send data to the client.
 
+### onClientAbort
+
+```js
+fastify.addHook('onClientAbort', (request, reply, done) => {
+  // Some code
+  done()
+})
+```
+Or `async/await`:
+```js
+fastify.addHook('onClientAbort', async (request, reply) => {
+  // Some code
+  await asyncMethod()
+})
+```
+`onClientAbort` is useful if you need to monitor the if the client aborts the request
+(if the `request.raw.aborted` property is set to true).
+The `onClientAbort` hook is executed when a client closes the connection before
+the entire response has been received. Therefore, you will not be able to send
+data to the client.
```

Comment:
We should update the lifecyle.md file as well

Reply:
See https://github.com/fastify/fastify/pull/4582#discussion_r1105524593

I do not consider this hook as part of lifecycle.